filebeat/filebeat.reference.yml

```diff
@@ -2034,7 +2034,8 @@ logging.files:
 # Defines if the HTTP endpoint is enabled.
 #http.enabled: false
 
-# The HTTP endpoint will bind to this hostname or IP address. It is recommended to use only localhost.
+# The HTTP endpoint will bind to this hostname, IP address or unix socket. It is recommended
+# to use only localhost.
```

Comment:
As far as the recommendation goes, what if it just said `If binding to an IP address it is recommended to use only localhost.`?

Reply:
@cachedout I think the wording is fine, but I presume @andrewkroh question was more should we default to npipe or Unix socket instead of TCP for the HTTP endpoint.

Reply:
My vote would be to default to the unix socket/npipe (starting 8.0.0 with deprecation warnings in 7.x, if possible) since its more secure (can only accept connections from local machine) and slightly faster than TCP. If users need the HTTP API to be callable from outside the local machine, they can take the deliberate step of configuring that.

Reply:
@ycombinator @cachedout What about the "beta" warning when starting the API?

Reply:
You mean this line?

```
2019-09-13T08:52:45.092-0700    WARN    [cfgwarn]       api/server.go:34        EXPERIMENTAL: Metrics endpoint is enabled.
```

If so, I vote for it to be taken out. As you said in another comment, this API has been around for a while now.

Reply:
I've removed the beta label but I didn't add a deprecation or changed the defaults.